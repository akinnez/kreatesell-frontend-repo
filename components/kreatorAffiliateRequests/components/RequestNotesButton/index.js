import Image from "next/image";
import { Button } from "antd";
import ReadImg from "public/images/note_read.png";
import UnreadImg from "public/images/note_unread.png";
import styles from "./index.module.scss";

const RequestNotesButton = ({ notes, noteFlag, showNoteModal }) => (
  <Button
    onClick={() => showNoteModal(notes)}
    shape="round"
    className={`${styles.notes__btn} ${
      noteFlag === "Unread"
        ? styles["notes__btn--unread"]
        : styles["notes__btn--read"]
    }`}
  >
    {noteFlag === "Unread" ? (
      <Image src={UnreadImg} alt="Unread Image" width={18} height={18} />
    ) : (
      <Image src={ReadImg} alt="Read Image" width={18} height={18} />
    )}
  </Button>
);

export default RequestNotesButton;
