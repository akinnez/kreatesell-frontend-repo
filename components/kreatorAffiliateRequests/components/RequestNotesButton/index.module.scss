@use 'public/css/colors';

.notes__btn {
	width: 40px;
	height: 40px;
	padding: 0;
	justify-content: center;
	border: none;
	border-radius: 8px;
	box-shadow: none;
	background-color: transparent;
}

.notes__btn--unread:hover {
	background-color: colors.$blue-100;
}

.notes__btn--read:hover {
	background-color: colors.$base-green-300;
}
