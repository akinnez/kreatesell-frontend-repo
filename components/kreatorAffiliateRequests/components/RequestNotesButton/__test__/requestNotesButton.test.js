import RequestNotesButton from "..";
import renderer from "react-test-renderer";

const mockProps = {
  notes: [".%&*", "^%$4"],
  showNoteModal: jest.fn(),
};

describe("RequestNotesButton : ", () => {
  it("renders a snapshot of <RequestNotesButton /> with mock props when noteFlag is not supplied", () => {
    const el = renderer.create(<RequestNotesButton {...mockProps} />).toJSON();
    expect(el).toMatchSnapshot();
  });
  it("renders a snapshot of <RequestNotesButton /> with mock props when noteFlag is  supplied", () => {
    const el = renderer
      .create(<RequestNotesButton {...mockProps} noteFlag="Unread" />)
      .toJSON();
    expect(el).toMatchSnapshot();
  });
});
