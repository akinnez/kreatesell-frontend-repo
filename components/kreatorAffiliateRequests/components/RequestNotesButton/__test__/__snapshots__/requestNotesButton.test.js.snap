// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestNotesButton :  renders a snapshot of <RequestNotesButton /> with mock props when noteFlag is  supplied 1`] = `
<button
  className="ant-btn ant-btn-round notes__btn notes__btn--unread"
  onClick={[Function]}
  type="button"
>
  <img
    src="[jest.find('../utils')]"
  />
</button>
`;

exports[`RequestNotesButton :  renders a snapshot of <RequestNotesButton /> with mock props when noteFlag is not supplied 1`] = `
<button
  className="ant-btn ant-btn-round notes__btn notes__btn--read"
  onClick={[Function]}
  type="button"
>
  <img
    src="[jest.find('../utils')]"
  />
</button>
`;
